### PR TITLE
Drive to ball

### PR DESCRIPTION
**Pull request by organelleV**
Robot tested code for toggling cameras and driving to ball

### CODE REVIEW

**Review by KevinH45 — APPROVED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -29,6 +29,25 @@
...
 import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.robot.commands.CameraDriveCommand;
+import frc.robot.NetworkTables.NetworkTablesBase;
+import frc.robot.commands.AutoShoot;
+import frc.robot.commands.CMD_ShooterManualRPM;
+import frc.robot.commands.CMD_canalThrough;
+import frc.robot.commands.CMD_changeSetpoint;
+import frc.robot.commands.CanalZeroToOneBottom;
+import frc.robot.subsystems.IndexSubsystem;
+import frc.robot.subsystems.Climber;
```
Comment (KevinH45) [inline src/main/java/frc/robot/RobotContainer.java]: in robot container this wall of imports is cleaned up using:
import frc.robot.commands.* and import frc.robot.subsystems.*

**Review by KevinH45 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (organelleV) [reply src/main/java/frc/robot/RobotContainer.java]: fixed it

**Review by KevinH45 — APPROVED** [review]

**Review by Spenceroni-Pepperoni — APPROVED** [review]

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -59,6 +61,9 @@
         private CanalSubsystem canal = new CanalSubsystem();
         private Climber climber = new Climber();
         private Limelight limelight = new Limelight();
+        private NetworkTablesBase networkTables = new NetworkTablesBase();
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/RobotContainer.java]: Comment out this line.

File: src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+/**
+ * Manages the canal subsystem, runs the front and back belts.
+ */
+public class CameraDriveSubsystem extends SubsystemBase {
+
+    public boolean direction = true; // true is forward, false is backward
+    public static boolean forward = true;
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java]: Should be `final`.

File: src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    public static boolean backward = false;
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java]: Unused variable.

File: src/main/java/frc/robot/subsystems/SUB_CameraData.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+/**
+ * Manages the canal subsystem, runs the front and back belts.
+ */
+public class SUB_CameraData extends SubsystemBase {
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/subsystems/SUB_CameraData.java]: Class is a duplicate of `CameraDriveSubsystem`. Remove one of them.

File: src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  /** Creates a new CanalSubsystem. */
+  public CameraDriveSubsystem() {
+
+  }
+  
+  @Override
+  public void periodic() {
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java:28]: Add method javadoc.

File: src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    // This method will be called once per scheduler run
+    if (direction == forward) {
+      SmartDashboard.putString("direction", "front");
+    } else {
+      SmartDashboard.putString("direction", "back");
+    }
+  }
+
+  public double getX(){
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java:40]: Add method javadoc.

File: src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    if (direction){
+      return SmartDashboard.getNumber("front_ball_x", 0);
+    } else {
+      return SmartDashboard.getNumber("back_ball_x", 0);
+    }
+  }
+
+    public double getY(){
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java:51]: Add method javadoc.

File: src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        if (direction){
+        return SmartDashboard.getNumber("front_ball_y", 0);
+        } else {
+        return SmartDashboard.getNumber("back_ball_y", 0);
+        }
+    }
+
+  public void toggleDirection() {
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/subsystems/CameraDriveSubsystem.java:62]: Add method javadoc.

File: src/main/java/frc/robot/commands/CameraDriveCommand.java [hunk 10]
```diff
@@ -9,24 +9,31 @@
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.subsystems.Drivetrain;
 import java.lang.Math;
+import frc.robot.subsystems.CameraDriveSubsystem;
 
 public class CameraDriveCommand extends CommandBase {
```
Comment (liam-a-wms) [inline src/main/java/frc/robot/commands/CameraDriveCommand.java:21]: Add class javadoc.

**Review by liam-a-wms — CHANGES_REQUESTED** [review]

**Review by liam-a-wms — APPROVED** [review]

**Pull request merged**